src/deep-security/lib/IdentityProvider.js

```diff
@@ -68,15 +68,50 @@ export class IdentityProvider {
       throw new IdentityProviderMismatchException(providerName, identityMetadata.provider);
     }
 
-    if (!identityMetadata.access_token || !identityMetadata.tokenExpirationTime) {
+    this._providers = providers;
+    this._name = providerDomain;
+    this._fillFromIdentityMetadata(providerName, identityMetadata);
```

Comment:
- Please change it to smth like normalizeIdentityMetadata and define all class properties inside constructor